torch/_decomp/decompositions.py

```diff
@@ -785,6 +785,25 @@ def diagonal_backward(
     return torch.diagonal_scatter(grad_input, grad_output, offset, dim1, dim2)
 
 
+@register_decomposition(aten.block_diag)
+@out_wrapper()
+def block_diag(tensors: List[Tensor]):
```

Comment:
This should be defined as a reference in `torch/_refs`, and you can add a `PythonRefInfo` in `torch/testing/_internal/common_methods_invocations.py` to generate tests for it.